### PR TITLE
Add buttons

### PR DESCRIPTION
**Pull request by yunpeng1234**
image
This how the buttons looks like as of now 

#94

### CODE REVIEW

**Comment by yunpeng1234** [comment]
Changed to 1 component with variant fields.
`green-solid green-outline black-solid black-outline`

File: components/Buttons.tsx [hunk 1]
```diff
@@ -8,35 +8,10 @@ type Props = {
   isLoading?: boolean;
   onClick?: () => any;
   icon: ReactNode;
+  variant: String;
```
Comment (kevinchua6) [inline components/Buttons.tsx]: btw this variant should be an enum right? and String should be small letter iirc

File: components/Buttons.tsx [hunk 2]
```diff
@@ -8,7 +8,7 @@ type Props = {
   isLoading?: boolean;
   onClick?: () => any;
   icon: ReactNode;
-  variant: String;
+  variant: 'green-solid' | 'green-outline' | 'black-solid' | 'black-outline';
```
Comment (kevinchua6) [inline components/Buttons.tsx:11]: oh nvm u changed it alr

**Review by MorningLit — CHANGES_REQUESTED** [review]

File: components/Buttons.tsx [hunk 3]
```diff
@@ -0,0 +1,83 @@
+import { Center, HStack, Spinner, Text } from '@chakra-ui/react';
+import React, { Component, ReactNode } from 'react';
+import { JsxElement } from 'typescript';
+
+type Props = {
+  text?: String;
+  isDisabled?: boolean;
+  isLoading?: boolean;
+  onClick?: () => any;
+  icon: ReactNode;
```
[the remark this answers is not in the document]
Comment (MorningLit) [reply components/Buttons.tsx:10]: ignore my comment on the left and right side

**Comment by MorningLit** [comment]
image
try and follow the ui format as shown, specifically the size part

File: components/Buttons.tsx [hunk 4]
```diff
@@ -0,0 +1,83 @@
+import { Center, HStack, Spinner, Text } from '@chakra-ui/react';
+import React, { Component, ReactNode } from 'react';
+import { JsxElement } from 'typescript';
+
+type Props = {
+  text?: String;
```
[the remark this answers is not in the document]
Comment (MorningLit) [reply components/Buttons.tsx:6]: actually lets remove text as a prop, we will take in children props and place our children within the <button>tags
eg. <button>{children}</button>

**Comment by yunpeng1234** [comment]
> image try and follow the ui format as shown, specifically the size part

Sorry but what is ghost button

File: components/Buttons.tsx [hunk 5]
```diff
@@ -0,0 +1,55 @@
+import { Center, HStack, Spinner, Text } from '@chakra-ui/react';
+import React, { Component, ReactNode } from 'react';
+import { JsxElement } from 'typescript';
```
Comment (kevinchua6) [inline components/Buttons.tsx:3]: Not used, remove this thanks!

File: components/Buttons.tsx [hunk 6]
```diff
@@ -0,0 +1,55 @@
+import { Center, HStack, Spinner, Text } from '@chakra-ui/react';
```
Comment (kevinchua6) [inline components/Buttons.tsx:1]: Remove HStack, Text and Component imports as they are unused

File: components/Buttons.tsx [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+    if (variant == 'black-solid') {
+      toShow = <Spinner color='#FFFFFF' />;
+    } else {
+      toShow = <Spinner />;
+    }
+  } else {
+    toShow = children;
+  }
+  const styling = getConfig(variant);
+  console.log(styling);
```
Comment (kevinchua6) [inline components/Buttons.tsx:25]: remove

File: components/Buttons.tsx [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+  return (
+    <button className={styling} disabled={isDisabled} onClick={onClick}>
+      <Center>{toShow}</Center>
+    </button>
+  );
+};
+
+const getConfig = (variant: string) => {
```
Comment (kevinchua6) [inline components/Buttons.tsx:33]: extract out type ('green-solid' | 'green-outline' | 'black-solid' | 'black-outline') on line 10 into a type Variants and use it as a type here

File: components/Buttons.tsx [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+  isDisabled?: boolean;
+  isLoading?: boolean;
+  onClick?: React.MouseEventHandler;
+  variant?: 'green-solid' | 'green-outline' | 'black-solid' | 'black-outline';
+  children?: ReactNode;
+};
+const CustomButton = ({ variant = 'green-solid', isDisabled = false, isLoading = false, onClick = () => {}, children = null }: Props) => {
+  let toShow;
+  if (isLoading) {
+    if (variant == 'black-solid') {
```
Comment (kevinchua6) [inline components/Buttons.tsx:16]: 3 equals

File: components/Buttons.tsx [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+    } else {
+      toShow = <Spinner />;
+    }
+  } else {
+    toShow = children;
+  }
+  const styling = getConfig(variant);
+  console.log(styling);
+  return (
+    <button className={styling} disabled={isDisabled} onClick={onClick}>
```
Comment (kevinchua6) [inline components/Buttons.tsx:27]: gj using tailwind

File: components/Buttons.tsx [hunk 11]
```diff
@@ -0,0 +1,55 @@
+import { Center, HStack, Spinner, Text } from '@chakra-ui/react';
+import React, { Component, ReactNode } from 'react';
+import { JsxElement } from 'typescript';
+
+type Props = {
```
Comment (kevinchua6) [inline components/Buttons.tsx:5]: add a className prop as well, so we can inject height, width properties etc into the button

eg
image

**Review by kevinchua6 — CHANGES_REQUESTED** [review]
Good job on the PR, just some changes needed

**Review by kevinchua6 — APPROVED** [review]

**Pull request merged**